chia/cmds/wallet_funcs.py

```diff
@@ -720,7 +721,17 @@ async def take_offer(
         royalty_asset_dict: Dict[Any, Tuple[Any, uint16]] = {}
         for royalty_asset_id in nft_coin_ids_supporting_royalties_from_offer(offer):
             if royalty_asset_id.hex() in offered:
-                percentage, address = await get_nft_royalty_percentage_and_address(royalty_asset_id, wallet_client)
+                eve_spend = next(
+                    (cs for cs in offer.coin_spends() if cs.coin.parent_coin_info == royalty_asset_id), None
+                )
+                if eve_spend:
```

Comment:
We prefer explicit checks against `None` there's been too many subtle bugs in this regard.  We should probably get it into a linter or something.